src/HealthChecks.AzureStorage/AzureBlobStorageHealthCheck.cs

```diff
@@ -9,20 +10,32 @@ namespace HealthChecks.AzureStorage
     public class AzureBlobStorageHealthCheck
         : IHealthCheck
     {
+        private readonly TokenCredential _azureCredential;
+        private readonly Uri _blobServiceUri;
+
         private readonly string _connectionString;
         private readonly string _containerName;
+
         public AzureBlobStorageHealthCheck(string connectionString, string containerName = default)
         {
             _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
             _containerName = containerName;
         }
+
+        public AzureBlobStorageHealthCheck(Uri blobServiceUri, TokenCredential credential, string containerName = default)
+        {
+            _blobServiceUri = blobServiceUri ?? throw new ArgumentNullException(nameof(blobServiceUri));
+            _azureCredential = credential ?? throw new ArgumentNullException(nameof(credential));
+            _containerName = containerName;
+        }
+
         public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
         {
             try
             {
-                var blobServiceClient = new BlobServiceClient(_connectionString);
+                var blobServiceClient = GetBlobServiceClient();

```

Comment:
This client should also be stored. Probably just do this and the other clients in the constructor. They are pretty lightweight and, if arguments are all valid, really shouldn't fail. Or at least do a lazy-init.

Reply:
Wait for response on my comment above about singleton healthchecks